solr/solr-ref-guide/modules/configuration-guide/pages/update-request-processors.adoc

```diff
@@ -337,6 +337,12 @@ Documents processed prior to the offender are indexed by Solr; documents followi
 +
 Alternatively, the processor offers a "permissive" mode (`permissiveMode=true`) which skips the offending document and logs a warning, but doesn't abort the remainder of the batch or return an error to users.
 
+{solr-javadocs}/core/org/apache/solr/update/processor/NumFieldLimitingUpdateRequestProcessorFactory.html[NumFieldLimitingUpdateRequestProcessorFactory]:: Fails update requests once a core has exceeded a configurable "maximum" number of fields.
++
+Solr performance can degrade and even become unstable if cores accumulate too many (e.g. more than 500) fields.  The "NumFieldLimiting" URP is offered as a safeguard that helps users notice potentially-dangerous schema design and/or mis-use of dynamic fields, before these performance and stability problems would manifest.
+Note that the field count an index reports can be influenced by deleted (but not yet purged) documents, and may vary from replica to replica.
+In order to avoid these sort of discrepancies between replicas, use of this URP should almost always precede DistributedUpdateProcessor in when running in SolrCloud mode.
```

Comment:
This advice surprised me.  I guess why not, as only the first node receiving the batch will evaluate this logic.  It may/may-not be the leader but whatever.

Reply:
Well, the URP certainly can't go _after_ DistributedUpdateProcessor for the reason laid out in the docs - you don't want some subset of your replicas rejecting certain docs because their segments have them disagree on the current field count.

Putting it _before_ DUP also has downsides, as you pointed out.  The field count check is going to happen on that somewhat arbitrary "first node", instead of a the more authoritative "leader".  And that adds wonkiness - if replicas disagree on field-count, this means you could get into situations where a given update could either succeed or fail depending on whether it gets lucky or not with its "first node".  (Hopefully this isn't a big deal though, since the URP isn't trying to be an exact limit - but it's still ugly and potentially confusing.)

This was what earlier versions of this PR were trying to avoid by gating the field-count check so that it only happened for docs that were being processed by their shard leader.